### ISSUE TITLE
letting custom backend handle the Appcenter Data Sync

### ISSUE THREAD

**Issue by Behrad-iz** [microsoft/appcenter#400]
like azure mobile API, allow data sync to be handled in private API that written in .net core web API instead of directly handled by CosmosDB.

so developers can decide what type of database and logic they are going to have despite bothering mobile about implementation.

**Comment by Zakeelm** [comment]
Hey there! 

Thanks for the feature request. This is definitely an interesting scenario. I would like to hear more. Can you elaborate on exactly what you would like to accomplish and what type of database/logic you're referring to?

**Comment by fileman** [comment]
I think @Behrad-iz refer to the azure mobile apps controller on server side.
With current AppCenter Data, the logic goes into mobile app, while with old azure mobile apps we can change the logic inside controller without update client app.

**Comment by Behrad-iz** [comment]
@Zakeelm 

i mean something like

**Comment by Zakeelm** [comment]
Thanks for the explanation (both of you)! With the link, i'm assuming you're referring to the server SDK piece right? Besides client app updating, are there any additional reasons why our current implementation may be a blocker for you? 

Since this is such a new product and we're in preview, i'm interested in investigating any way we can improve the developer experience. Want to make sure we're solving the right problems as we iterate.

**Comment by fileman** [comment]
It should be useful for a project with centralized service back-end with business logic in this tier (azure functions, webapi with/out odata, etc...) with different type of client (Xamarin, WebApp, SPA, WPF, etc...).

The main problem for me is that are missing offline support for list of document and public collection are read only, but I'll open a specific issue for those.

**Comment by Zakeelm** [comment]
That makes sense. I can definitely see why this would be a useful scenario. Our team is definitely going to investigate this further.

But that's perfect! We can discuss those further in those issues. Luckily, we do have items in our roadmap that should assist with those blockers

I'll leave this thread open for a while for visibility in case other want to discuss this item

**Comment by fileman** [comment]
I've seen two items in the updated (1 day after my comment :-) ) [roadmap](https://github.com/Microsoft/appcenter/wiki/Roadmap) and these could be a great solutions:
- Collaborate on collection documents -> multiple user can edit documents in collection
- Real-time document storage updates -> can prevent concurrency problem with LWW specially when Cosmos DB [partial document update](https://feedback.azure.com/forums/263030-azure-cosmos-db/suggestions/6693091-be-able-to-do-partial-updates-on-document) goes into GA

**Comment by alexcnichols** [comment]
Hi @Behrad-iz and @fileman -- Beyond the improvements to Data itself that @Zakeelm has planned, we're also starting research around adding compute backend functionality to App Center more generally. If you're willing to chat more, head over to #993 for more information.

**Comment by Zakeelm** [comment]
@Behrad-iz  and @fileman  Thank you so much for your requests. We announced this week in a [blog post]( https://devblogs.microsoft.com/appcenter/app-center-mbaas-retirement/) that we’re retiring the Auth and Data services later this year and recommending customers transition to Azure Active Directory B2C and Azure Cosmos DB instead. Let me know if you have any additional questions about this transition and i'll make sure to get them answered! Closing this issue.
